lib/optimize/CommonsChunkPlugin.js

```diff
@@ -24,153 +24,304 @@ The available options are:
 	minSize: number`);
 		}
 
-		if(Array.isArray(options) || typeof options === "string") {
-			options = {
-				name: options
+		const normalizedOptions = this.normalizeOptions(options);
+
+		this.chunkNames = normalizedOptions.chunkNames;
+		this.filenameTemplate = this.filenameTemplate;
+		this.minChunks = normalizedOptions.minChunks;
+		this.selectedChunks = normalizedOptions.selectedChunks;
+		this.async = normalizedOptions.async;
+		this.minSize = normalizedOptions.minSize;
+		this.ident = normalizedOptions.ident;
+	}
+
+	normalizeOptions(options) {
+		if(Array.isArray(options)) {
+			return {
+				chunkNames: options,
+			};
+		}
+
+		if(typeof options === "string") {
+			return {
+				chunkNames: [options],
 			};
 		}
-		this.chunkNames = options.name || options.names;
-		this.filenameTemplate = options.filename;
-		this.minChunks = options.minChunks;
-		this.selectedChunks = options.chunks;
-		if(options.children) this.selectedChunks = false;
-		this.async = options.async;
-		this.minSize = options.minSize;
-		this.ident = __filename + (nextIdent++);
+
+		// if "children" is set, set selectedChunks to false instead of specified chunks
+		// TODO: why? :P
+		const selectedChunks = options.children ? false : options.chunks;
+		const chunkNames = options.name ? [options.name] : options.names;
+		return {
+			chunkNames: chunkNames,
+			filenameTemplate: options.filename,
+			minChunks: options.minChunks,
+			selectedChunks: selectedChunks,
+			async: options.async,
+			minSize: options.minSize,
+			ident: __filename + (nextIdent++),
+		};
 	}
+
+	getCommonChunks(allChunks, compilation, chunkNames, selectedChunks, async) {
+		const asyncOrNoSelectedChunk = selectedChunks === false || async;
```

Comment:
instead of `selectedChunks === false` you can refactor this to `this.options.chilren`